internal/core/src/common/Array.h

```diff
@@ -438,19 +438,42 @@ class Array {
     int size_ = 0;
     std::vector<uint64_t> offsets_{};
```

Comment:
1. why not still keep offset ? Didn't see a strong reason to do so.
2. why not change offset to unint32?